user/reference/overview.md

```diff
@@ -37,6 +47,28 @@ The following table summarizes the differences between the virtual environments:
 | Memory           | 7.5 GB                                | 4 GB max                          | 7.5 GB                               | 4 GB                         |
 | Cores            | ~2, bursted                           | 2                                 | ~2, bursted                          | 2                            |
 
+## What infrastructure is my environment running on?
+
+Usually, knowing the virtualization environment charactistics from the [table above](#Virtualisation-Environment-vs-Operating-System) is sufficient.
+
+But, if you do need more detail, you have one of these two questions:
+
+* you want to see what infrastructure a [finished build](#For-a-finished-build) ran on.
+* you want to determine what infrastructure a [particular `.travis.yml` configuration](#For-a-particular-.travis.yml-configuration) will run on.
+
+### For a finished build
+
+To see what infrastructure a finished build ran on, look at the *hostname* at the top of the build log:
+
+![Infrastructure shown in hostname](/images/ui/what-infrastructure.png "Infrastructure shown in hostname")
+
+if it contains:
+
+* `ec2`, as in the previous example → the build ran in a container-based environment on Amazon EC2.
+* `gce` → the build ran in a sudo-enabled environment on Google Cloud Engine.
+* `jupiter` → macOS
+
+### For a particular .travis.yml configuration
```

Comment:
So in a very generic way we can say: 

- a language specified + no `os:` key  = Linux
also `os: linux`
  - `services: docker`,  `sudo: required`, `sudo: true` or any sudo commands in your .travis.yml file will send you to the fully virtualized Linux environment.

   - `sudo: false` or `no key specified` and no sudo commands in a `.travis.yml` - sends you to the container-based Linux environment

- `os: osx` or any of the `osx_image:` keys send you to the macOS infrastructure